core/services/validators.go

```diff
@@ -392,36 +408,119 @@ func ValidatedOracleSpec(tomlString string) (spec offchainreporting.OracleSpec,
 	for _, k := range m.Undecoded() {
 		err = multierr.Append(err, errors.Errorf("unrecognised key: %s", k))
 	}
+	if !m.IsDefined("isBootstrapPeer") {
+		return spec, errors.New("isBootstrapPeer is not defined")
+	}
 	if spec.IsBootstrapPeer {
-		return validatedBootstrapSpec(m, spec)
+		if err := validateBootstrapSpec(m, spec); err != nil {
+			return spec, err
+		}
+	} else if err := validateNonBootstrapSpec(m, spec); err != nil {
+		return spec, err
+	}
+	// TODO: expose these various constants from libocr so they are defined in one place.
+	if time.Duration(spec.ObservationTimeout) < 1*time.Millisecond || time.Duration(spec.ObservationTimeout) > 20*time.Second {
+		return spec, errors.Errorf("require 1ms <= observation timeout <= 20s")
+	}
+	if time.Duration(spec.BlockchainTimeout) < 1*time.Millisecond || time.Duration(spec.ObservationTimeout) > 20*time.Second {
+		return spec, errors.Errorf("require 1ms <= blockchain timeout <= 20s ")
+	}
+	if time.Duration(spec.ContractConfigTrackerPollInterval) < 15*time.Second || time.Duration(spec.ContractConfigTrackerPollInterval) > 120*time.Second {
+		return spec, errors.Errorf("require 15s <= contract config tracker poll interval <= 120s ")
 	}
-	return
+	if time.Duration(spec.ContractConfigTrackerSubscribeInterval) < 2*time.Minute || time.Duration(spec.ContractConfigTrackerSubscribeInterval) > 5*time.Minute {
+		return spec, errors.Errorf("require 2m <= contract config subscribe interval <= 5m ")
+	}
+	if spec.ContractConfigConfirmations < 2 || spec.ContractConfigConfirmations > 10 {
+		return spec, errors.Errorf("require 2 <= contract config confirmations <= 10 ")
+	}
+	return spec, nil
 }
 
-var bootstrapKeys = map[string]struct{}{
-	`type`:                                   struct{}{},
-	`schemaVersion`:                          struct{}{},
-	`contractAddress`:                        struct{}{},
-	`p2pPeerID`:                              struct{}{},
-	`p2pBootstrapPeers`:                      struct{}{},
-	`isBootstrapPeer`:                        struct{}{},
-	`blockchainTimeout`:                      struct{}{},
-	`contractConfigTrackerSubscribeInterval`: struct{}{},
-	`contractConfigTrackerPollInterval`:      struct{}{},
-	`contractConfigConfirmations`:            struct{}{},
+// Parameters that must be explicitly set by the operator.
```

Comment:
For what

Reply:
Not sure what you mean - just required parameters needed to specify an OCR job